src/cpp/src/group_beam_searcher.cpp

```diff
@@ -452,9 +449,10 @@ EncodedResults beam_search(ov::InferRequest& lm,
         ) {
             results.scores.push_back(beam->get().score);
             results.tokens.push_back(std::move(beam->get().tokens));
+            selected_beams.push_back(beam->get().global_beam_idx);
         }
     }
-    return results;
+    return {results, selected_beams};
```

Comment:
should we return a single element? E.g. beam idx with highest score.
I don't see how multiple idx could be used on successive iterations

Reply:
changed to return a single selected_beam